jme3-core/src/main/java/com/jme3/anim/tween/action/BlendSpace.java

```diff
@@ -1,10 +1,84 @@
+/*
+ * Copyright (c) 2009-2023 jMonkeyEngine
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ * * Redistributions of source code must retain the above copyright
+ *   notice, this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright
+ *   notice, this list of conditions and the following disclaimer in the
+ *   documentation and/or other materials provided with the distribution.
+ *
+ * * Neither the name of 'jMonkeyEngine' nor the names of its contributors
+ *   may be used to endorse or promote products derived from this software
+ *   without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+ * TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
 package com.jme3.anim.tween.action;
 
+/**
+ * Controls the blending between 2 successive actions in a {@link BlendAction} by adjusting blending weight value
+ * {@link BlendSpace#getWeight()} during application runtime based on an arbitrary value {@link BlendSpace#setValue(float)}.
+ * 
+ * <p>
+ * Notes:
+ * <li> Blending is the action of mixing between 2 successive animation {@link BlendableAction}s by interpolating their transforms and 
+ * then applying them on a {@link HasLocalTransform} object, the {@link BlendSpace} provides this blending action with a blend weight value. </li>
+ * <li> The blend weight is the value for the interpolation for the target transforms. </li>
+ * <li> The blend weight value should lie in this interval [0, 1]. </li>
+ * </p>
+ * 
+ * <p>
+ * Different blending weight case scenarios managed by {@link BlendAction} internally:
+ * <li> In case of (0 < Blending weight < 1), the blending is executed each update among 2 actions, the first action will use 
+ * a blend value of 1 and the second action will use the blend space weight as a value for the interpolation. </li>
+ * <li> In case of (Blending weight <= 0), the blending hasn't started yet, only the first action will be interpolated at (weight = 1). </li>
```

Comment:
I hope I am correct about this; because the code is not clear about the negative values!

So, Are the negative values treated the same as the {0}?

Reply:
BlendSpace should never return a negative weight. You only need to document that `getWeight()` must return a value in [0, 1].

Reply:
BlendAction has a conditional statements that handle different blend weight values for the user, I guess this should be documented in the root origin of blend weight which is the blend space?

Reply:
I am trying to document the blend weight values behavior.

Reply:
Okay, I still donot fully agree with you, I think every side effect should be documented, as a user if I am going to build a new blend space on my game, it would be nice if the API tells me everything about the blend weight behavior.

Note: if you think that there is a defective behavior regarding the negative values, then it should be corrected before we go over documenting this API.

Reply:
I think there is a missing check here:

https://github.com/jMonkeyEngine/jmonkeyengine/blob/7025310b8f767e0b5956468f9ae2ef6ca8dd8604/jme3-core/src/main/java/com/jme3/anim/tween/action/BlendableAction.java#L93

In the above method, we can throw an illegal argument exception if it is not in [0, 1].

Like what I did here:

https://github.com/jMonkeyEngine/jmonkeyengine/blob/7025310b8f767e0b5956468f9ae2ef6ca8dd8604/jme3-core/src/main/java/com/jme3/anim/tween/action/BlendableAction.java#L120-L126

Reply:
Which is better, throwing a business exception or printing a warning message denoting that the interpolation is skipped? 

I know we must be consistent, but I am just asking.

Reply:
I think we should discuss these missing checks on forums, to keep this PR clean from unrelated contents.

Reply:
>interpolation is skipped

I do not how we can skip it, but it feels wrong to me. We should either throw an exception or clamp it into [0, 1]. 

And 99% of the time, people will use LinererBlendSpace which will never return a value other than [0, 1].

If someone wants to create a BlendSpace that returns a negative weight that is their issue.

Reply:
According to `BlendAction#collectTransform`, negative values are treated the same as zero, so I guess they are skipped.

But you are right regarding people's blame 🤣, I agree throwing an exception will be the righteous choice.

Reply:
We can then document it here as "Negative values are not allowed".

Reply:
>"Negative values are not allowed".

and values greater than 1, or just saying "weight must be in [0, 1]" which indicates both.

Reply:
Looks great, will you work on the PR to add this restriction, please?

Reply:
Yes, I can submit a PR.

Reply:
I have added the requested changes, if you think that this will satisfy the new behavior, please resolve this conversation.